presto-docs/src/main/sphinx/functions/array.rst

```diff
@@ -102,6 +102,11 @@ Array Functions
                                        -1,
                                        IF(cardinality(x) = cardinality(y), 0, 1))); -- [[1, 2], [2, 3, 1], [4, 2, 1, 4]]
 
+.. function:: array_sum(array(T)) -> T
+
+    Returns the sum of all non-null elements of the ``array``. If there is no non-null elements, returns ``0``.
+    ``T`` must be coercible to ``double`` or ``bigint``.
+
```

Comment:
May be mention here that it's similar to the SUM aggregation.

Reply:
Added 
```
The behavior is similar to aggregation function :func:`sum`.
```

Reply:
It should be `T can only be double or bigint` or you should write it as `array_sum(array(T)) -> double / bigint` If the array is `array(int)` the return type is not `int`. Or just write them as 2 functions.

Reply:
Description updated.
```
``T`` must be coercible to ``double``.
    Returns ``bigint`` if T is coercible to ``bigint``. Otherwise, returns ``double``.
```